### ISSUE TITLE
Rule Proposal: no-tabs

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6079]
From [disallowTabs].

This rule will disallow a use of tabs.
I guess this rule is really similar to [no-irregular-whitespace] rule.
So the option of [no-irregular-whitespace] may be better. Thoughts?

```js
{
    "no-tabs": "error"
}
```

[disallowTabs]: http://jscs.info/rule/disallowTabs
[no-irregular-whitespace]: http://eslint.org/docs/rules/no-irregular-whitespace

**Comment by platinumazure** [comment]
This is covered by `indent` and whichever rule has a smart-tabs option for variable alignment, I think?

**Comment by gyandeeps** [comment]
I have this rule in my plugin: https://github.com/gyandeeps/eslint-plugin-ideal/blob/master/lib/rules/no-tabs-in-file.js
It can be ported here if needed.

**Comment by mysticatea** [comment]
@platinumazure `smart-tabs` allows a use of tabs.
@gyandeeps sounds good. JSCS rule seems to check tabs in whole a file, too.

**Comment by platinumazure** [comment]
I hadn't considered the idea that maybe we want to check tabs in comments,
so I agree this should maybe be a separate rule. Ideally this new rule
would not interfere with indent (perhaps by not checking beginning of line).

**Comment by nzakas** [comment]
Per TSC meeting, we've accepted the `no-tabs` proposal.

**Comment by gyandeeps** [comment]
Working on this, require suggestions on

1. Error message
2. Should we activate this rule for eslint project
